javascript/javascript3/week3/script.js

```diff
@@ -0,0 +1,240 @@
+class Circle{
+    constructor(x,y,r,startAngle,endAngle,fillColor, ctx){
+        this.x = x;
+        this.y = y;
+        this.r = r;
+        this.startAngle = startAngle;
+        this.endAngle = endAngle;
+        this.fillColor = fillColor;
+        this.ctx = ctx;
+    }
+
+    draw(){
+        this.ctx.beginPath();
+        this.ctx.moveTo(this.x,this.y);
+        this.ctx.arc(this.x, this.y, this.r, this.startAngle,this.endAngle);
+        this.ctx.fillStyle = this.fillColor;
+        this.ctx.fill();
+    }
+}
+
+//Lets make some art using classes
+const canvas1 = document.getElementById("canvas1");
+const ctx1 = canvas1.getContext("2d");
+const circles = [
+    new Circle(282,214,15,0,360,"green",ctx1),
+    new Circle(125,302,90,0,360,"red",ctx1),
+    new Circle(130,94,40,0,360,"blue",ctx1),
+    new Circle(400,460,200,0,360,"yellow",ctx1),
+    new Circle(456,159,38,0,360,"orange",ctx1)
+]
+
+circles.forEach(c=>{
+    c.draw();
+})
+
+
+
+//Now lets make art
+const canvas2 = document.getElementById("canvas2");
+const ctx2 = canvas2.getContext("2d");
+
+function randomInt(min, max) {
+    min = Math.ceil(min);
+    max = Math.floor(max);
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+function getRandomColor() {
+    const letters = '0123456789ABCDEF';
+    let color = '#';
+    for (var i = 0; i < 6; i++) {
+      color += letters[Math.floor(Math.random() * 16)];
+    }
+    return color;
+}
+
+const maxCircles = 1000;
+let counter = 0;
+function intervalCircles(){
+    const interv = setInterval(()=>{
+        counter++;
+        const c = new Circle(randomInt(0,500),randomInt(0,500),randomInt(5,30),0,360,getRandomColor(),ctx2);
+        c.draw();
+        if(counter >= maxCircles){
+            clearInterval(interv);
+        }
+    },100)
+}
+
+intervalCircles();
+
+//Classmates
+const classmates = document.getElementById("classmates");
+
+function renderRepositories(username){
```

Comment:
The task here is to have one function that returns a Promise for the repository data for one user, and then one function that takes a list of such user data and renders it to the page.
It's a very common pattern to learn, to separate data loading from page rendering.